src/Files.App/Views/Properties/HashesPage.xaml

```diff
@@ -26,14 +27,74 @@
 
 	<Grid x:Name="RootGrid">
 
+		<!--  Hash Comparison Section  -->
 		<Grid
-			x:Name="HashesListGrid"
+			x:Name="HashComparisonGrid"
 			Margin="12"
 			VerticalAlignment="Top"
 			Background="{ThemeResource CardBackgroundFillColorDefaultBrush}"
 			BorderBrush="{ThemeResource CardStrokeColorDefaultBrush}"
 			BorderThickness="1"
 			CornerRadius="4">
+			<Grid.RowDefinitions>
+				<RowDefinition Height="Auto" />
+				<RowDefinition Height="Auto" />
+			</Grid.RowDefinitions>
+			<Grid.ColumnDefinitions>
+				<ColumnDefinition Width="*" />
+				<ColumnDefinition Width="Auto" />
+			</Grid.ColumnDefinitions>
+
+			<!--  Hash Input  -->
+			<Grid
+				Grid.Row="0"
+				Grid.Column="0"
+				Margin="12">
+				<Grid.ColumnDefinitions>
+					<ColumnDefinition Width="*" />
+				</Grid.ColumnDefinitions>
+
+				<TextBox
+					x:Name="HashInputTextBox"
+					Grid.Column="0"
+					PlaceholderText="{helpers:ResourceString Name=EnterHashToCompare}"
+					Text="{x:Bind HashesViewModel.HashInput, Mode=TwoWay}">
+					<interactivity:Interaction.Behaviors>
+						<interactivity:EventTriggerBehavior EventName="TextChanged">
+							<interactivity:InvokeCommandAction Command="{x:Bind HashesViewModel.HashInputTextChangedCommand}" />
+						</interactivity:EventTriggerBehavior>
+					</interactivity:Interaction.Behaviors>
+				</TextBox>
+			</Grid>
+
+			<!--  Compare File Button  -->
+			<Button
+				x:Name="CompareFileButton"
+				Grid.Row="0"
+				Grid.Column="1"
+				Margin="12"
+				Command="{x:Bind HashesViewModel.CompareFileCommand}"
+				Content="{helpers:ResourceString Name=CompareFile}" />
+		</Grid>
+
+		<InfoBar
+			x:Name="HashMatchInfoBar"
+			Grid.Row="1"
+			Grid.Column="0"
+			Margin="12,75,12,12"
```

Comment:
Instead of a top margin, it's best practice to use a separate row. This will also fix the issue where a large gap is displayed when the banner is hidden.

Reply:
Sure, I'll implement the UI niceties after proper MVVM functionality is implemented and works

Reply:
I can't do this until I move the row definitions to RootGrid but I'm encountering an issue when doing so.

Reply:
This issue has been resolved.